dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CloudInstanceResource.java

```diff
@@ -0,0 +1,97 @@
+package io.dockstore.webservice.resources;
+
+import java.util.List;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.codahale.metrics.annotation.Timed;
+import io.dockstore.webservice.core.CloudInstance;
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.jdbi.CloudInstanceDAO;
+import io.dropwizard.auth.Auth;
+import io.dropwizard.hibernate.UnitOfWork;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.enums.ParameterIn;
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.apache.http.HttpStatus;
+import org.hibernate.SessionFactory;
+
+import static io.dockstore.webservice.resources.ResourceConstants.OPENAPI_JWT_SECURITY_DEFINITION_NAME;
+
+@Path("/cloudInstances")
+@Produces(MediaType.APPLICATION_JSON)
+@Tag(name = "Cloud Instances")
+public class CloudInstanceResource implements AuthenticatedResourceInterface {
+    private CloudInstanceDAO cloudInstanceDAO;
+
+    public CloudInstanceResource(SessionFactory sessionFactory) {
+        this.cloudInstanceDAO = new CloudInstanceDAO(sessionFactory);
+
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("/")
+    @Operation(operationId = "getCloudInstances", summary = "Get all known public cloud instances")
+    @ApiResponse(responseCode = HttpStatus.SC_OK
+            + "", description = "Got all known public cloud instances", content = @Content(array = @ArraySchema(schema = @Schema(implementation = CloudInstance.class))))
+    public List<CloudInstance> getCloudInstances() {
+        return this.cloudInstanceDAO.findAllWithoutUser();
+    }
+
+    @DELETE
+    @Timed
+    @UnitOfWork
+    @Path("/{cloudInstanceId}")
+    @RolesAllowed({ "admin" })
+    @Operation(operationId = "deleteCloudInstance", summary = "Delete a public cloud instance, admin only", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = HttpStatus.SC_OK + ""),
```

Comment:
In Natalie's PR, didn't she say void returns 204?

Reply:
Ok, assuming 204 then